controllers/install/executor_controller.go

```diff
@@ -405,28 +460,73 @@ func (r *ExecutorReconciler) createClusterRole(executor *installv1alpha1.Executo
 func (r *ExecutorReconciler) createClusterRoleBinding(
 	executor *installv1alpha1.Executor,
 	clusterRole *rbacv1.ClusterRole,
-	serviceAccount *corev1.ServiceAccount,
+	serviceAccountName string,
 ) *rbacv1.ClusterRoleBinding {
-	serviceAccountName := executor.Spec.CustomServiceAccount
-	if serviceAccountName == "" {
-		serviceAccountName = serviceAccount.Name
-	}
 	clusterRoleBinding := rbacv1.ClusterRoleBinding{
 		ObjectMeta: metav1.ObjectMeta{Name: executor.Name, Labels: AllLabels(executor.Name, executor.Labels)},
 		Subjects: []rbacv1.Subject{{
 			Kind:      "ServiceAccount",
 			Name:      serviceAccountName,
-			Namespace: serviceAccount.Namespace,
+			Namespace: executor.Namespace,
 		}},
 		RoleRef: rbacv1.RoleRef{
 			Kind:     "ClusterRole",
-			APIGroup: "rbac.authorization.k8s.io",
+			APIGroup: rbacv1.GroupName,
 			Name:     clusterRole.Name,
 		},
 	}
 	return &clusterRoleBinding
 }
 
+func (r *ExecutorReconciler) createAdditionalClusterRoleBindings(executor *installv1alpha1.Executor, serviceAccountName string) []*rbacv1.ClusterRoleBinding {
```

Comment:
Should this be preemptively moved to common_helpers.go -- and take []rbacv1.ClusterRoleBinding arg instead of specific CR?

Reply:
this one is a bit customized from what I have seen in the Armada helm charts, if we do that, we will need to change the Helm charts.